sdk/parents/azure-client-sdk-parent/pom.xml

```diff
@@ -107,7 +106,35 @@
     <jacoco.min.linecoverage>0.40</jacoco.min.linecoverage>
     <jacoco.min.branchcoverage>0.30</jacoco.min.branchcoverage>
     <jacoco.skip.coverage.check>false</jacoco.skip.coverage.check>
+
+    <!-- This property determines the relative path from a pom.xml to the eng/ folder. -->
     <relative.path.to.eng.folder>../../..</relative.path.to.eng.folder>
+
+    <!-- This property determines the max heap size of the Surefire JVM. By default 3GB, or 3096MB, is used. -->
+    <surefireXmx>3096m</surefireXmx>
+
+    <!-- This property configures the max heap size of the Surefire JVM. -->
+    <surefireJvmXmx>-Xmx${surefireXmx}</surefireJvmXmx>
```

Comment:
can this collide with https://github.com/Azure/azure-sdk-for-java/blob/da6337e07cf937ab23ce2e1dafd65529334c4e3a/sdk/storage/azure-storage-blob/pom.xml#L236-L238
 ?

Reply:
Yes, this can collide with that. The thought behind the new pattern is to override `<surefireXmx>3096m</surefireXmx>` to a new value in the child POM, so in this case it would be `<surefireXmx>6g</surefireXmx>` in `azure-storage-blob/pom.xml`. This will then be rolled into the `<surefireJvmXmx>` value and passed into the `maven-surefire-plugin` `argLine`.

Overall, the changes are to use more composition-esque styling by overriding property values instead of overriding plugin configurations (as these have shown to be much more brittle).